srtcore/group.cpp

```diff
@@ -1351,10 +1354,11 @@ int CUDTGroup::sendBroadcast(const char* buf, int len, SRT_MSGCTRL& w_mc)
         sendstates.push_back(cstate);
     }
 
-    if (curseq != SRT_SEQNO_NONE)
+    if (nextseq != SRT_SEQNO_NONE)
     {
-        HLOGC(gslog.Debug, log << "grp/sendBroadcast: updating current scheduling sequence %" << curseq);
-        m_iLastSchedSeqNo = curseq;
+        HLOGC(gslog.Debug,
+              log << "grp/sendBroadcast: $" << id() << ": updating current scheduling sequence %" << nextseq);
+        m_iLastSchedSeqNo = nextseq;
```

Comment:
Maybe just using the `curseq` so that there is no need to add another variable `nextseq`?

```c++
m_iLastSchedSeqNo = CSeqNo::incseq(curseq);
```

Reply:
I'm afraid not, if the message across multiple packets,  `nextseq` is not equals to `incseq(curseq)`.

Reply:
Ah, you are right! I forgot you are using broadcast groups with the message mode :)
If a message is formed by several packets, then after a message is submitted for sending, the next sequence number is indeed `curseq + msg_size_in_pkts`.

Reply:
Hm. After `sendmsg2(..)` the `w_mc.pktseq` is set to the next scheduled pkt seqno. Meaning that `curseq = w_mc.pktseq` although quite not obviously, but seems to be setting the same value as `d->ps->core().schedSeqNo()`.
https://github.com/Haivision/srt/blob/master/srtcore/buffer.cpp#L297

Reply:
I'm afraid not,
```
        m_pSndBuffer->addBuffer(data, size, (w_mctrl));
        m_iSndNextSeqNo = w_mctrl.pktseq;
        w_mctrl.pktseq = seqno;
```
after  `addBuffer()`, `w_mctrl.pktseq` will be  overridden to `seqno`.

Reply:
You are right again 😄